solr/core/src/java/org/apache/solr/search/FunctionQParser.java

```diff
@@ -91,6 +106,18 @@ public boolean getParseToEnd() {
 
   @Override
   public Query parse() throws SyntaxError {
+    return new FunctionQuery(parseAsValueSource());
+  }
+
+  /**
+   * Parses as a ValueSource, not a Query. <em>NOT</em> intended to be called by {@link
+   * ValueSourceParser#parse(FunctionQParser)}; it's intended for general code that has a {@link
+   * QParser} but actually wants to parse a ValueSource.
+   *
+   * @return A {@link VectorValueSource} for multiple VS, otherwise just the single VS.
+   */
+  @Override
+  public ValueSource parseAsValueSource() throws SyntaxError {
```

Comment:
During dev I once used the existing method `parseValueSource()` but that provided problematic as there are different expectations between VSP callers and QP callers.